### ISSUE TITLE
getUserFromClient is not defined 

### ISSUE THREAD

**Issue by barisvelioglu** [oauthjs/node-oauth2-server#364]
**grant.js line 293**

```
  return this.model.getUserFromClient(clientId, clientSecret,
      function (err, user) {
    if (err) return done(error('server_error', false, err));
    if (!user) {
      return done(error('invalid_grant', 'Client credentials are invalid'));
    }

    self.user = user;
    done();
  });
```

This method returns undefined when I try to get access_token with client_credentials. I check the code but I couldnt find the implementation of it.

**Comment by maxtruxa** [comment]
This is one of the `model` functions you have to implement yourself.

See [2.4.1 README](https://github.com/oauthjs/node-oauth2-server/tree/b36a06b445ad0a676e6175d68a8bd0b2f3353dbf#getuserfromclient-clientid-clientsecret-callback).

For example, if you don't care about the `user`, or just to try it out:

```
app.oauth = oauthserver({
  model: {
    // ...
    getUserFromClient: function(clientId, clientSecret) {
        return {};
    }
  },
  grants: ['password'],
  debug: true
});
```

**Comment by AlbertHambardzumyan** [comment]
@maxtruxa You mentioned `password` grant type in your example. The only usage for this function is `client_credentials`.

@barisvelioglu I have implemented without caring about `user` and its looks like:
`this.getUserFromClient = (clientId, clientSecret, callback) => {
        callback(null, true);
    };`

**Comment by maxtruxa** [comment]
@AlbertHambardzumyan You're right. IIRC I just copied the basic example from the 2.x README (that's where `password` came from) and added the handler.
